pkg/proxy/nftables/proxier.go

```diff
@@ -991,10 +1097,19 @@ func (proxier *Proxier) syncProxyRules() {
 	// Below this point we will not return until we try to write the nftables rules.
 	//
 
+	// The value of proxier.needFullSync may change before the defer funcs run, so
+	// we need to keep track of whether it was set at the *start* of the sync.
+	tryPartialSync := !proxier.needFullSync
+
 	// Keep track of how long syncs take.
 	start := time.Now()
 	defer func() {
 		metrics.SyncProxyRulesLatency.Observe(metrics.SinceInSeconds(start))
+		if tryPartialSync {
+			metrics.SyncPartialProxyRulesLatency.Observe(metrics.SinceInSeconds(start))
```

Comment:
need to update `pkg/proxy/metrics/metrics.go` to register this metric for nftables